// react-pdf-tailwind/dist/index.mjs
// src/index.ts
import defaultTheme from "tailwindcss/defaultTheme";

// src/colors.ts
var tailwindColors = {
  slate: {
    50: "#f8fafc",
    100: "#f1f5f9",
    200: "#e2e8f0",
    300: "#cbd5e1",
    400: "#94a3b8",
    500: "#64748b",
    600: "#475569",
    700: "#334155",
    800: "#1e293b",
    900: "#0f172a",
    950: "#020617"
  },
  gray: {
    50: "#f9fafb",
    100: "#f3f4f6",
    200: "#e5e7eb",
    300: "#d1d5db",
    400: "#9ca3af",
    500: "#6b7280",
    600: "#4b5563",
    700: "#374151",
    800: "#1f2937",
    900: "#111827",
    950: "#030712"
  },
  zinc: {
    50: "#fafafa",
    100: "#f4f4f5",
    200: "#e4e4e7",
    300: "#d4d4d8",
    400: "#a1a1aa",
    500: "#71717a",
    600: "#52525b",
    700: "#3f3f46",
    800: "#27272a",
    900: "#18181b",
    950: "#09090b"
  },
  neutral: {
    50: "#fafafa",
    100: "#f5f5f5",
    200: "#e5e5e5",
    300: "#d4d4d4",
    400: "#a3a3a3",
    500: "#737373",
    600: "#525252",
    700: "#404040",
    800: "#262626",
    900: "#171717",
    950: "#0a0a0a"
  },
  stone: {
    50: "#fafaf9",
    100: "#f5f5f4",
    200: "#e7e5e4",
    300: "#d6d3d1",
    400: "#a8a29e",
    500: "#78716c",
    600: "#57534e",
    700: "#44403c",
    800: "#292524",
    900: "#1c1917",
    950: "#0c0a09"
  },
  red: {
    50: "#fef2f2",
    100: "#fee2e2",
    200: "#fecaca",
    300: "#fca5a5",
    400: "#f87171",
    500: "#ef4444",
    600: "#dc2626",
    700: "#b91c1c",
    800: "#991b1b",
    900: "#7f1d1d",
    950: "#450a0a"
  },
  orange: {
    50: "#fff7ed",
    100: "#ffedd5",
    200: "#fed7aa",
    300: "#fdba74",
    400: "#fb923c",
    500: "#f97316",
    600: "#ea580c",
    700: "#c2410c",
    800: "#9a3412",
    900: "#7c2d12",
    950: "#431407"
  },
  amber: {
    50: "#fffbeb",
    100: "#fef3c7",
    200: "#fde68a",
    300: "#fcd34d",
    400: "#fbbf24",
    500: "#f59e0b",
    600: "#d97706",
    700: "#b45309",
    800: "#92400e",
    900: "#78350f",
    950: "#451a03"
  },
  yellow: {
    50: "#fefce8",
    100: "#fef9c3",
    200: "#fef08a",
    300: "#fde047",
    400: "#facc15",
    500: "#eab308",
    600: "#ca8a04",
    700: "#a16207",
    800: "#854d0e",
    900: "#713f12",
    950: "#422006"
  },
  lime: {
    50: "#f7fee7",
    100: "#ecfccb",
    200: "#d9f99d",
    300: "#bef264",
    400: "#a3e635",
    500: "#84cc16",
    600: "#65a30d",
    700: "#4d7c0f",
    800: "#3f6212",
    900: "#365314",
    950: "#1a2e05"
  },
  green: {
    50: "#f0fdf4",
    100: "#dcfce7",
    200: "#bbf7d0",
    300: "#86efac",
    400: "#4ade80",
    500: "#22c55e",
    600: "#16a34a",
    700: "#15803d",
    800: "#166534",
    900: "#14532d",
    950: "#052e16"
  },
  emerald: {
    50: "#ecfdf5",
    100: "#d1fae5",
    200: "#a7f3d0",
    300: "#6ee7b7",
    400: "#34d399",
    500: "#10b981",
    600: "#059669",
    700: "#047857",
    800: "#065f46",
    900: "#064e3b",
    950: "#022c22"
  },
  teal: {
    50: "#f0fdfa",
    100: "#ccfbf1",
    200: "#99f6e4",
    300: "#5eead4",
    400: "#2dd4bf",
    500: "#14b8a6",
    600: "#0d9488",
    700: "#0f766e",
    800: "#115e59",
    900: "#134e4a",
    950: "#042f2e"
  },
  cyan: {
    50: "#ecfeff",
    100: "#cffafe",
    200: "#a5f3fc",
    300: "#67e8f9",
    400: "#22d3ee",
    500: "#06b6d4",
    600: "#0891b2",
    700: "#0e7490",
    800: "#155e75",
    900: "#164e63",
    950: "#083344"
  },
  sky: {
    50: "#f0f9ff",
    100: "#e0f2fe",
    200: "#bae6fd",
    300: "#7dd3fc",
    400: "#38bdf8",
    500: "#0ea5e9",
    600: "#0284c7",
    700: "#0369a1",
    800: "#075985",
    900: "#0c4a6e",
    950: "#082f49"
  },
  blue: {
    50: "#eff6ff",
    100: "#dbeafe",
    200: "#bfdbfe",
    300: "#93c5fd",
    400: "#60a5fa",
    500: "#3b82f6",
    600: "#2563eb",
    700: "#1d4ed8",
    800: "#1e40af",
    900: "#1e3a8a",
    950: "#172554"
  },
  indigo: {
    50: "#eef2ff",
    100: "#e0e7ff",
    200: "#c7d2fe",
    300: "#a5b4fc",
    400: "#818cf8",
    500: "#6366f1",
    600: "#4f46e5",
    700: "#4338ca",
    800: "#3730a3",
    900: "#312e81",
    950: "#1e1b4b"
  },
  violet: {
    50: "#f5f3ff",
    100: "#ede9fe",
    200: "#ddd6fe",
    300: "#c4b5fd",
    400: "#a78bfa",
    500: "#8b5cf6",
    600: "#7c3aed",
    700: "#6d28d9",
    800: "#5b21b6",
    900: "#4c1d95",
    950: "#2e1065"
  },
  purple: {
    50: "#faf5ff",
    100: "#f3e8ff",
    200: "#e9d5ff",
    300: "#d8b4fe",
    400: "#c084fc",
    500: "#a855f7",
    600: "#9333ea",
    700: "#7e22ce",
    800: "#6b21a8",
    900: "#581c87",
    950: "#3b0764"
  },
  fuchsia: {
    50: "#fdf4ff",
    100: "#fae8ff",
    200: "#f5d0fe",
    300: "#f0abfc",
    400: "#e879f9",
    500: "#d946ef",
    600: "#c026d3",
    700: "#a21caf",
    800: "#86198f",
    900: "#701a75",
    950: "#4a044e"
  },
  pink: {
    50: "#fdf2f8",
    100: "#fce7f3",
    200: "#fbcfe8",
    300: "#f9a8d4",
    400: "#f472b6",
    500: "#ec4899",
    600: "#db2777",
    700: "#be185d",
    800: "#9d174d",
    900: "#831843",
    950: "#500724"
  },
  rose: {
    50: "#fff1f2",
    100: "#ffe4e6",
    200: "#fecdd3",
    300: "#fda4af",
    400: "#fb7185",
    500: "#f43f5e",
    600: "#e11d48",
    700: "#be123c",
    800: "#9f1239",
    900: "#881337",
    950: "#4c0519"
  }
};

// src/properties.ts
var negativeProperties = [
  "zIndex",
  "top",
  "right",
  "bottom",
  "left",
  "translate",
  "scale",
  "rotate",
  "order",
  "margin"
];
function isNegativeProperty(key) {
  return typeof key === "string" && negativeProperties.includes(key);
}
var scaledProperties = [
  "borderRadius",
  "borderWidth",
  "flexBasis",
  "fontFamily",
  "fontSize",
  "fontWeight",
  "gap",
  "height",
  "inset",
  "letterSpacing",
  "lineHeight",
  "margin",
  "maxHeight",
  "maxWidth",
  "minHeight",
  "minWidth",
  "objectPosition",
  "opacity",
  "order",
  "padding",
  "rotate",
  "scale",
  "textIndent",
  "transformOrigin",
  "translate",
  "width",
  "zIndex"
];
function isScaledProperty(key) {
  return typeof key === "string" && scaledProperties.includes(key);
}
var exactUtilities = {
  // Layout
  flex: { display: "flex" },
  hidden: { display: "none" },
  "object-contain": { objectFit: "contain" },
  "object-cover": { objectFit: "cover" },
  "object-fill": { objectFit: "fill" },
  "object-none": { objectFit: "none" },
  "object-scale-down": { objectFit: "scale-down" },
  "overflow-hidden": { overflow: "hidden" },
  absolute: { position: "absolute" },
  relative: { position: "relative" },
  // Flexbox
  "flex-row": { flexDirection: "row" },
  "flex-row-reverse": { flexDirection: "row-reverse" },
  "flex-col": { flexDirection: "column" },
  "flex-col-reverse": { flexDirection: "column-reverse" },
  "flex-wrap": { flexWrap: "wrap" },
  "flex-wrap-reverse": { flexWrap: "wrap-reverse" },
  "flex-nowrap": { flexWrap: "nowrap" },
  "flex-1": { flex: "1 1 0%" },
  "flex-auto": { flex: "1 1 auto" },
  "flex-initial": { flex: "0 1 auto" },
  "flex-none": { flex: "none" },
  grow: { flexGrow: 1 },
  "grow-0": { flexGrow: 0 },
  shrink: { flexShrink: 1 },
  "shrink-0": { flexShrink: 0 },
  "justify-start": { justifyContent: "flex-start" },
  "justify-end": { justifyContent: "flex-end" },
  "justify-center": { justifyContent: "center" },
  "justify-between": { justifyContent: "space-between" },
  "justify-around": { justifyContent: "space-around" },
  "justify-evenly": { justifyContent: "space-evenly" },
  "content-start": { alignContent: "flex-start" },
  "content-end": { alignContent: "flex-end" },
  "content-center": { alignContent: "center" },
  "content-between": { alignContent: "space-between" },
  "content-around": { alignContent: "space-around" },
  "items-start": { alignItems: "flex-start" },
  "items-end": { alignItems: "flex-end" },
  "items-center": { alignItems: "center" },
  "items-baseline": { alignItems: "baseline" },
  "items-stretch": { alignItems: "stretch" },
  "self-auto": { alignSelf: "auto" },
  "self-start": { alignSelf: "flex-start" },
  "self-end": { alignSelf: "flex-end" },
  "self-center": { alignSelf: "center" },
  "self-baseline": { alignSelf: "baseline" },
  "self-stretch": { alignSelf: "stretch" },
  // Typography
  italic: { fontStyle: "italic" },
  "not-italic": { fontStyle: "normal" },
  "text-left": { textAlign: "left" },
  "text-center": { textAlign: "center" },
  "text-right": { textAlign: "right" },
  "text-justify": { textAlign: "justify" },
  "text-inherit": { color: "inherit" },
  "text-current": { color: "currentColor" },
  "text-transparent": { color: "transparent" },
  "text-black": { color: "black" },
  "text-white": { color: "white" },
  underline: { textDecoration: "underline" },
  "decoration-solid": { textDecorationStyle: "solid" },
  "decoration-double": { textDecorationStyle: "double" },
  "decoration-dotted": { textDecorationStyle: "dotted" },
  "decoration-dashed": { textDecorationStyle: "dashed" },
  "decoration-wavy": { textDecorationStyle: "wavy" },
  "decoration-inherit": { textDecorationColor: "inherit" },
  "decoration-current": { textDecorationColor: "currentColor" },
  "decoration-transparent": { textDecorationColor: "transparent" },
  "decoration-black": { textDecorationColor: "black" },
  "decoration-white": { textDecorationColor: "white" },
  "line-through": { textDecoration: "line-through" },
  "no-underline": { textDecoration: "none" },
  uppercase: { textTransform: "uppercase" },
  lowercase: { textTransform: "lowercase" },
  capitalize: { textTransform: "capitalize" },
  truncate: {
    overflow: "hidden",
    textOverflow: "ellipsis"
  },
  "text-ellipsis": { textOverflow: "ellipsis" },
  // Backgrounds
  "bg-inherit": { backgroundColor: "inherit" },
  "bg-current": { backgroundColor: "currentColor" },
  "bg-transparent": { backgroundColor: "transparent" },
  "bg-black": { backgroundColor: "black" },
  "bg-white": { backgroundColor: "white" },
  // Borders
  "border-solid": { borderStyle: "solid" },
  "border-dashed": { borderStyle: "dashed" },
  "border-dotted": { borderStyle: "dotted" },
  "border-inherit": { borderColor: "inherit" },
  "border-current": { borderColor: "currentColor" },
  "border-transparent": { borderColor: "transparent" },
  "border-black": { borderColor: "black" },
  "border-white": { borderColor: "white" },
  "border-x-inherit": {
    borderLeftColor: "inherit",
    borderRightColor: "inherit"
  },
  "border-x-current": {
    borderLeftColor: "currentColor",
    borderRightColor: "currentColor"
  },
  "border-x-transparent": {
    borderLeftColor: "transparent",
    borderRightColor: "transparent"
  },
  "border-x-black": { borderLeftColor: "black", borderRightColor: "black" },
  "border-x-white": { borderLeftColor: "white", borderRightColor: "white" },
  "border-y-inherit": {
    borderTopColor: "inherit",
    borderBottomColor: "inherit"
  },
  "border-y-current": {
    borderTopColor: "currentColor",
    borderBottomColor: "currentColor"
  },
  "border-y-transparent": {
    borderTopColor: "transparent",
    borderBottomColor: "transparent"
  },
  "border-y-black": { borderTopColor: "black", borderBottomColor: "black" },
  "border-y-white": { borderTopColor: "white", borderBottomColor: "white" },
  "border-t-inherit": { borderTopColor: "inherit" },
  "border-t-current": { borderTopColor: "currentColor" },
  "border-t-transparent": { borderTopColor: "transparent" },
  "border-t-black": { borderTopColor: "black" },
  "border-t-white": { borderTopColor: "white" },
  "border-r-inherit": { borderRightColor: "inherit" },
  "border-r-current": { borderRightColor: "currentColor" },
  "border-r-transparent": { borderRightColor: "transparent" },
  "border-r-black": { borderRightColor: "black" },
  "border-r-white": { borderRightColor: "white" },
  "border-b-inherit": { borderBottomColor: "inherit" },
  "border-b-current": { borderBottomColor: "currentColor" },
  "border-b-transparent": { borderBottomColor: "transparent" },
  "border-b-black": { borderBottomColor: "black" },
  "border-b-white": { borderBottomColor: "white" },
  "border-l-inherit": { borderLeftColor: "inherit" },
  "border-l-current": { borderLeftColor: "currentColor" },
  "border-l-transparent": { borderLeftColor: "transparent" },
  "border-l-black": { borderLeftColor: "black" },
  "border-l-white": { borderLeftColor: "white" }
};
var utilityPatterns = {
  // Layout
  object: "objectPosition",
  top: "top",
  right: "right",
  bottom: "bottom",
  left: "left",
  z: "zIndex",
  // Flexbox
  basis: "flexBasis",
  flex: "flex",
  "gap-x": ["gap", "columnGap"],
  "gap-y": ["gap", "rowGap"],
  gap: "gap",
  grow: "flexGrow",
  shrink: "flexShrink",
  order: "order",
  // Spacing
  m: "margin",
  mx: ["margin", ["marginLeft", "marginRight"]],
  my: ["margin", ["marginTop", "marginBottom"]],
  ml: ["margin", "marginLeft"],
  mr: ["margin", "marginRight"],
  mt: ["margin", "marginTop"],
  mb: ["margin", "marginBottom"],
  p: "padding",
  px: ["padding", ["paddingLeft", "paddingRight"]],
  py: ["padding", ["paddingTop", "paddingBottom"]],
  pl: ["padding", "paddingLeft"],
  pr: ["padding", "paddingRight"],
  pt: ["padding", "paddingTop"],
  pb: ["padding", "paddingBottom"],
  // Sizing
  w: "width",
  "min-w": "minWidth",
  "max-w": "maxWidth",
  h: "height",
  "min-h": "minHeight",
  "max-h": "maxHeight",
  // Typography
  leading: "lineHeight",
  tracking: "letterSpacing",
  indent: "textIndent",
  // Backgrounds
  bg: "backgroundColor",
  // Effects
  opacity: "opacity",
  // Transforms
  origin: "transformOrigin"
};

// src/utils.ts
var PT_PER_PX = 1;
var PT_PER_REM = 12;
function round(value) {
  return Math.round(value * 1e6) / 1e6;
}
function isNumeric(value) {
  if (value.startsWith(".")) {
    value = `0${value}`;
  }
  return Number(value).toString() === value;
}
function capitalize(string) {
  return `${string.charAt(0).toUpperCase()}${string.slice(1).toLowerCase()}`;
}
function rem(value, ptPerRem = PT_PER_REM) {
  return round(ptPerRem * value);
}
function px(value) {
  return round(PT_PER_PX * value);
}

// src/index.ts
function themeResolver(name) {
  return defaultTheme[name] ?? {};
}
function createTw(config, options) {
  const theme = { ...defaultTheme, ...config };
  const colors = { ...tailwindColors, ...config.colors ?? {} };
  const cache = {};
  function transformValue(value, property, isNegative) {
    if (value === void 0) {
      return void 0;
    }
    if (typeof value !== "string" && typeof value !== "number") {
      console.warn(
        "Invalid value type passed to transformValue",
        value,
        property
      );
      return void 0;
    }
    const sign = isNegative ? -1 : 1;
    if (typeof value === "number") {
      return sign * value;
    }
    switch (property) {
      case "lineHeight":
        if (value.endsWith("rem")) {
          return sign * Number(value.replace("rem", ""));
        }
        return sign * Number(value);
      default:
        if (value.endsWith("px")) {
          return px(sign * Number(value.replace("px", "")));
        }
        if (value.endsWith("rem")) {
          return rem(
            sign * Number(value.replace("rem", "")),
            options?.ptPerRem
          );
        }
        if (value.endsWith("em")) {
          return rem(sign * Number(value.replace("em", "")), options?.ptPerRem);
        }
        if (isNegative && property && isNegativeProperty(property)) {
          const suffix = ["deg", "%"].find((i) => value.endsWith(i));
          if (suffix) {
            return `${sign * Number(value.replace(suffix, ""))}${suffix}`;
          }
        }
        if (isNumeric(value)) {
          return sign * Number(value);
        }
        return value;
    }
  }
  function getCustomValue(value) {
    if (value.startsWith("[") && value.endsWith("]")) {
      return value.slice(1, value.length - 1).replaceAll("_", " ");
    }
    return void 0;
  }
  function isMultiProperty(value) {
    return Array.isArray(value);
  }
  function parseValue(value, property, isNegative) {
    const valueParts = value.split("-");
    const customValue = getCustomValue(value);
    if (customValue) {
      if (["#", "rgb", "hsl"].some((prefix) => customValue.startsWith(prefix))) {
        return {
          value: customValue,
          type: "color",
          isCustom: true
        };
      }
      if (["px", "rem"].some((suffix) => customValue.endsWith(suffix))) {
        return {
          value: transformValue(customValue, property, isNegative),
          type: "unit",
          isCustom: true
        };
      }
      return {
        value: transformValue(customValue, property, isNegative),
        type: "other",
        isCustom: true
      };
    }
    if (valueParts[0] && valueParts[0] in colors && property !== "fontWeight") {
      const color = colors[valueParts[0]];
      return {
        value: typeof color === "string" ? color : valueParts[1] && typeof valueParts[1] === "string" ? color?.[valueParts[1]] : void 0,
        type: "color",
        isCustom: false,
        additionalProperties: void 0
      };
    }
    if (valueParts.length === 0 || !property) {
      return {
        value: void 0
      };
    }
    const maybeScaledProperty = ["top", "right", "bottom", "left"].includes(
      property
    ) ? "inset" : property;
    if (isScaledProperty(maybeScaledProperty)) {
      const key = maybeScaledProperty;
      const themeProp = theme[key];
      let result;
      if (typeof themeProp === "function") {
        result = themeProp({ theme: themeResolver })?.[value];
      } else if (themeProp && value in themeProp) {
        result = themeProp?.[value];
      }
      if (!result) {
        result = theme.spacing[value];
      }
      if (isMultiProperty(result)) {
        const additionalProperties = result[1] && result[1] !== null && typeof result[1] === "object" ? Object.fromEntries(
          Object.entries(result[1]).map(([key2, value2]) => [
            key2,
            transformValue(value2, key2)
          ])
        ) : null;
        return {
          value: transformValue(result[0], property, isNegative),
          type: "unit",
          isCustom: false,
          ...additionalProperties ? { additionalProperties } : null
        };
      }
      return {
        value: transformValue(result, property, isNegative),
        type: "unit",
        isCustom: false
      };
    }
    return {
      value: void 0
    };
  }
  function parseUtility(className) {
    const modifierParts = className.split(":");
    const utilityStr = modifierParts[modifierParts.length - 1];
    if (utilityStr && utilityStr in exactUtilities) {
      return exactUtilities[utilityStr];
    }
    const isNegative = utilityStr ? utilityStr.startsWith("-") : false;
    const utilityParts = utilityStr ? utilityStr.slice(isNegative ? 1 : 0).split("-") : [];
    const matchingUtilityPatternKey = Object.keys(utilityPatterns).find(
      (key) => {
        const keyParts = key.split("-");
        const comparisonKey = utilityParts.slice(0, keyParts.length).join("-");
        return key === comparisonKey;
      }
    );
    if (matchingUtilityPatternKey) {
      const rawValue = className.split(`${matchingUtilityPatternKey}-`)[1];
      const pattern = utilityPatterns[matchingUtilityPatternKey];
      const property = Array.isArray(pattern) ? pattern[0] : pattern;
      const mappedProperties = Array.isArray(pattern) ? Array.isArray(pattern[1]) ? pattern[1] : [pattern[1]] : [pattern];
      if (!rawValue || isNegative && !isNegativeProperty(property)) {
        return void 0;
      }
      const { value, additionalProperties } = parseValue(
        rawValue,
        property,
        isNegative
      );
      return {
        ...Object.fromEntries(mappedProperties.map((prop) => [prop, value])),
        ...additionalProperties ?? null
      };
    }
    switch (utilityParts[0]) {
      case "inset": {
        const direction = ["x", "y"].find((i) => i === utilityParts[1]);
        const valueStr = utilityParts.slice(direction ? 2 : 1).join("-");
        const { value } = parseValue(valueStr, "inset", isNegative);
        switch (direction) {
          case "x":
            return {
              left: value,
              right: value
            };
          case "y":
            return {
              top: value,
              bottom: value
            };
          default:
            return {
              top: value,
              right: value,
              bottom: value,
              left: value
            };
        }
      }
      case "font": {
        const valueStr = utilityParts.slice(1).join("-");
        const customValue = getCustomValue(valueStr);
        if (customValue) {
          if (isNumeric(customValue)) {
            return {
              fontWeight: parseInt(customValue)
            };
          }
          return {
            fontFamily: customValue
          };
        }
        if (theme.fontFamily && valueStr in theme.fontFamily) {
          const { value: value2 } = parseValue(valueStr, "fontFamily");
          return {
            fontFamily: value2
          };
        }
        const { value } = parseValue(valueStr, "fontWeight");
        return {
          fontWeight: value
        };
      }
      case "text": {
        const valueStr = utilityParts.slice(1).join("-");
        const { value, additionalProperties, type } = parseValue(
          valueStr,
          "fontSize"
        );
        if (type === "color") {
          return { color: value };
        }
        return { fontSize: value, ...additionalProperties };
      }
      case "decoration": {
        const valueStr = utilityParts.slice(1).join("-");
        const { value, type } = parseValue(valueStr, "textDecorationColor");
        if (type === "color") {
          return {
            textDecorationColor: value
          };
        }
        return void 0;
      }
      case "rounded": {
        const direction = ["t", "r", "b", "l", "tl", "tr", "br", "bl"].find(
          (i) => i === utilityParts[1]
        );
        const valueStr = utilityParts.slice(direction ? 2 : 1).join("-");
        const { value } = parseValue(valueStr || "DEFAULT", "borderRadius");
        switch (direction) {
          case "t":
            return {
              borderTopLeftRadius: value,
              borderTopRightRadius: value
            };
          case "r":
            return {
              borderTopRightRadius: value,
              borderBottomRightRadius: value
            };
          case "b":
            return {
              borderBottomRightRadius: value,
              borderBottomLeftRadius: value
            };
          case "l":
            return {
              borderBottomLeftRadius: value,
              borderTopLeftRadius: value
            };
          case "tl":
            return {
              borderTopLeftRadius: value
            };
          case "tr":
            return {
              borderTopRightRadius: value
            };
          case "br":
            return {
              borderBottomRightRadius: value
            };
          case "bl":
            return {
              borderBottomLeftRadius: value
            };
          default:
            return {
              borderRadius: value
            };
        }
      }
      case "border": {
        const direction = ["x", "y", "t", "r", "b", "l"].find(
          (i) => i === utilityParts[1]
        );
        const valueStr = utilityParts.slice(direction ? 2 : 1).join("-");
        const { value, type } = parseValue(
          valueStr || "DEFAULT",
          "borderWidth"
        );
        const propertySuffix = capitalize(type === "color" ? "color" : "width");
        switch (direction) {
          case "x":
            return {
              [`borderLeft${propertySuffix}`]: value,
              [`borderRight${propertySuffix}`]: value
            };
          case "y":
            return {
              [`borderTop${propertySuffix}`]: value,
              [`borderBottom${propertySuffix}`]: value
            };
          case "t":
            return { [`borderTop${propertySuffix}`]: value };
          case "r":
            return { [`borderRight${propertySuffix}`]: value };
          case "b":
            return { [`borderBottom${propertySuffix}`]: value };
          case "l":
            return { [`borderLeft${propertySuffix}`]: value };
          default:
            return {
              [`border${propertySuffix}`]: value
            };
        }
      }
      case "scale": {
        const direction = ["x", "y"].find((i) => i === utilityParts[1]);
        const valueStr = utilityParts.slice(direction ? 2 : 1).join("-");
        const { value } = parseValue(valueStr, "scale", isNegative);
        switch (direction) {
          case "x":
            return {
              transform: `scaleX(${value})`
            };
          case "y":
            return {
              transform: `scaleY(${value})`
            };
          default:
            return {
              transform: `scale(${value})`
            };
        }
      }
      case "rotate": {
        const { value } = parseValue(
          utilityParts.slice(1).join("-"),
          "rotate",
          isNegative
        );
        return {
          transform: `rotate(${value})`
        };
      }
      case "translate": {
        const direction = ["x", "y"].find((i) => i === utilityParts[1]);
        const valueStr = utilityParts.slice(direction ? 2 : 1).join("-");
        const { value } = parseValue(valueStr, "translate", isNegative);
        switch (direction) {
          case "x":
            return {
              transform: `translateX(${value})`
            };
          case "y":
            return {
              transform: `translateY(${value})`
            };
          default:
            return {
              transform: `translate(${value})`
            };
        }
      }
    }
    return void 0;
  }
  function handleInvalidClassName(className) {
    console.warn(`react-pdf-tailwind: Invalid class "${className}"`);
  }
  return function(input) {
    const classNames = input.split(" ").map((i) => i.trim());
    return classNames.map((className) => {
      if (className in cache) {
        return cache[className];
      }
      const parsed = parseUtility(className);
      if (parsed && Object.values(parsed).every((v) => typeof v !== "undefined")) {
        cache[className] = parsed;
        return parsed;
      } else {
        handleInvalidClassName(className);
      }
      return void 0;
    }).reduce((acc, val) => {
      if (!val) {
        return acc;
      }
      if ("transform" in val) {
        const { transform, ...rest } = val;
        return {
          ...acc,
          ...transform ? { transform: [acc.transform ?? "", transform].join(" ").trim() } : null,
          ...rest
        };
      }
      return { ...acc, ...val };
    }, {});
  };
}
export {
  createTw
};
